GetStream/stream-chat-swift
# [CIS-795] Remove defaultAppearance Hook

# Motivation for PR:
- `DefaultAppearance` hook usage was ambigous and we agreed that we should delete the implementation because we cannot **restrict the usage of this method only to modify appearance** and not actually code for layout and other properties which could potentially lead to side effects because of the nature of `StreamChat._View` lifecycle. 

# What this PR does: 
- Removes `Customizable.defaultAppearance` method which was responsible for default `StreamChat` appearance. With this it also removes `defaultAppearance hook`  which was the root motivation for this PR. Instead of using `defaultAppearance` in our codebase, we switch to overriding `setUpAppearance` with `super.setUpAppearance` calls.
# How to test it
- The biggest prove should be the tests being green. Nevertheless always try to run the app and take a look if something is behaving strange or is different layout then before, if yes we should fix it and add test to it as it's not covered. 

Please check the code snippets and documentation if I should improve something or is not clear.
